crates/viewer/re_space_view_dataframe/src/view_query.rs

```diff
@@ -0,0 +1,495 @@
+            .find(|t| t.name() == &timeline_name)
+    });
+
+    let mut override_query = timeline.is_some();
+    let changed = ui.selectable_toggle(|ui| {
+        ui.selectable_value(&mut override_query, false, "Follow timeline")
+            .changed()
+            || ui
+                .selectable_value(&mut override_query, true, "Override")
+                .changed()
+    });
+
+    if changed {
+        if override_query {
+            let time_ctrl = ctx.rec_cfg.time_ctrl.read();
+            let timeline = time_ctrl.timeline();
+
+            // UX least surprising behavior: when switching from "follow" to "override", we ensure
+            // that the override configuration defaults to the current timeline configuration, so
+            // the table content remains stable.
+            property.save_blueprint_component(
+                ctx,
+                &components::Timeline::from(timeline.name().as_str()),
+            );
+            Query::save_mode_for_timeline(
+                ctx,
+                space_view_id,
+                timeline.name(),
+                &QueryMode::LatestAt {
+                    time: time_ctrl.time_int().unwrap_or(TimeInt::MAX),
+                },
+            )?;
+        } else {
+            property.reset_blueprint_component::<components::Timeline>(ctx);
+        }
+    }
+
+    if override_query {
+        override_ui(ctx, ui, state, space_view_id, &property)
+    } else {
+        Ok(())
+    }
+}
+
+fn override_ui(
+    ctx: &ViewerContext<'_>,
+    ui: &mut egui::Ui,
+    state: &dyn SpaceViewState,
+    space_view_id: SpaceViewId,
+    property: &ViewProperty<'_>,
+) -> Result<(), SpaceViewSystemExecutionError> {
+    let name = archetypes::DataframeQuery::name();
+    let Some(reflection) = ctx.reflection.archetypes.get(&name) else {
+        // The `ArchetypeReflectionMarker` bound should make this impossible.
+        re_log::warn_once!("Missing reflection data for archetype {name:?}.");
+        return Ok(());
+    };
+
+    let timeline = property
+        .component_or_empty::<components::Timeline>()?
+        .map(|t| t.timeline_name())
+        .and_then(|timeline_name| {
+            ctx.recording()
+                .timelines()
+                .find(|t| t.name() == &timeline_name)
+                .copied()
+        })
+        .unwrap_or(*ctx.rec_cfg.time_ctrl.read().timeline());
+    let timeline_name = timeline.name();
+
+    let blueprint_path =
+        entity_path_for_view_property(space_view_id, ctx.blueprint_db().tree(), name);
+    let query_context = QueryContext {
+        viewer_ctx: ctx,
+        target_entity_path: &blueprint_path,
+        archetype_name: Some(name),
+        query: ctx.blueprint_query,
+        view_state: state,
+        view_ctx: None,
+    };
+
+    let component_results = ctx.blueprint_db().latest_at(
+        ctx.blueprint_query,
+        &blueprint_path,
+        reflection.fields.iter().map(|field| field.component_name),
+    );
```

Comment:
I believe this is the one single location where I use reflection on `archetype::DataframeQuery`, which in turn requires the `Default` impls. 

I guess I could hard-code the component names, but that would be flaky against component rename and archetype expansion.